src/Devices/Kiosk/KioskDevice.cs

```diff
@@ -74,7 +84,7 @@ private Task<MethodResponse> ReceivePurchaseTicketResponse(MethodRequest methodR
             string result = "{\"result\":\"Executed direct method: " + methodRequest.Name + "\"}";
 
             // restart timer
-            purchaseEvent.Start();
+            this.eventScheduler.StartAll(); // TODO: need more elegant solution
```

Comment:
Please, remove the TODO, fix it, or create a work item for fixing it & link it here.

Reply:
removed and used a different approach. May revisit use of Observer pattern and reactive extensions at a later date if time allows.